File: 1cpp/Hexagon.cpp
//
// Created by olile on 10.04.24.
//

#include "Hexagon.h"

#include <stdexcept>
#include <cmath>

Hexagon::Hexagon(double a) noexcept(false) {
    if(a < 0){
        throw std::invalid_argument("żaden argument nie może być  ujemny.");
    }
    this->a = a;
}
double Hexagon::calcArea(){
    return this->a*this->a*sqrt(27)/2;
}
double Hexagon::calcCircumference(){
    return 6*this->a;
}
std::string Hexagon::info(){
    return "Hexagon";
}

File: 1java/1java/src/Hexagon.java
// Hexagon.java (Example implementation)
public class Hexagon implements Shape {
    private double a;

    public Hexagon(double a) {
        if (a < 0) {
            throw new IllegalArgumentException("Argument must not be negative.");
        }
        this.a = a;
    }

    @Override
    public double calcArea() {
        // Implement the area calculation for a regular hexagon
        // Return the actual area value.
        return a * a * Math.sqrt(27) / 2;
    }

    @Override
    public double calcCircumference() {
        return 6 * a;
    }

    @Override
    public String info() {
        return "Hexagon";
    }
}
